Add setting for optionally disabling ansible-runner process isolation

enpaul opened the pull request:
<!--- changelog-entry
# Fill in 'msg' below to have an entry automatically added to the next release changelog.
# Leaving 'msg' blank will not generate a changelog entry for this PR.
# Please ensure this is a simple (and readable) one-line string.
---
msg: "Add `RUNNER_PROCESS_ISOLATION` setting to configure containerized isolation via ansible-runner"
-->

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

Currently the `process_isolation` parameter for Ansible Runner is [hardcoded to `True`](https://github.com/ansible/awx/blob/devel/awx/main/tasks.py#L975). This update makes this setting configurable via a Django setting called `RUNNER_PROCESS_ISOLATION` to allow a user to disable containerized process isolation if they want to (the default of `True` remains unchanged). This is intended to support development, testing, and small-scale environments where using containerized execution environments is either not possible or unnecessary.

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 19.5.1.dev26+g3232b6b005
```

enpaul committed:
Add setting for optionally disabling ansible-runner process isolation

Add RUNNER_PROCESS_ISOLATION setting to make the process isolation behavior of
ansible-runner configurable to support small and testing deployments that don't
use managed execution environments.

Signed-off-by: Ethan Paul <24588726+enpaul@users.noreply.github.com>

AlanCoding commented on awx/main/tasks.py:
```diff
@@ -969,6 +969,9 @@ def build_execution_environment_params(self, instance, private_data_dir):
             return {}
 
+        if not settings.RUNNER_PROCESS_ISOLATION:
+            return {"process_isolation": False}
```
I can't imagine this would work right because project updates and other job types use `container_volume_mounts`... and I don't think those are compatible with setting this to False, but I'm not sure what the actual ansible-runner behavior would be in that case.

enpaul replied:
Hm, I believe you're correct that these options are incompatible with each. I'll check that now.

However, I think this can still work if the `container_volume_mounts` option is removed as well when this is false, since the job will be running directly under the parent process and so will have access to the project directory. I'll do some more testing to make sure this is the case though.

enpaul replied:
@AlanCoding I did confirm that this does work as I expect. The one thing to note is that the default value for `AWX_ISOLATION_BASE_PATH` doesn't work when this setting is set to `False` because container's don't share a `/tmp` location. However, by setting `AWX_ISOLATION_BASE_PATH` to a shared volume mounted in both containers this new setting worked exactly as intended.

enpaul committed:
Update CONTAINER_ROOT constant to support unisolated jobs

When job process isolation is disabled, the "container root" should be the
same as the project path on the work executor since no container is being
launched.
Update default pod spec container args to use CONTAINER_ROOT constant

Signed-off-by: Ethan Paul <24588726+enpaul@users.noreply.github.com>
